internal/user/repository/eventsourcing/model/user.go

```diff
@@ -0,0 +1,236 @@
+		ObjectRoot: user.ObjectRoot,
+		State:      model.UserState(user.State),
+	}
+	if user.Password != nil {
+		converted.Password = PasswordToModel(user.Password)
+	}
+	if user.Profile != nil {
+		converted.Profile = ProfileToModel(user.Profile)
+	}
+	if user.Email != nil {
+		converted.Email = EmailToModel(user.Email)
+	}
+	if user.Phone != nil {
+		converted.Phone = PhoneToModel(user.Phone)
+	}
+	if user.Address != nil {
+		converted.Address = AddressToModel(user.Address)
+	}
+	if user.InitCode != nil {
+		converted.InitCode = InitCodeToModel(user.InitCode)
+	}
+	if user.EmailCode != nil {
+		converted.EmailCode = EmailCodeToModel(user.EmailCode)
+	}
+	if user.PhoneCode != nil {
+		converted.PhoneCode = PhoneCodeToModel(user.PhoneCode)
+	}
+	if user.PasswordCode != nil {
+		converted.PasswordCode = PasswordCodeToModel(user.PasswordCode)
+	}
+	if user.OTP != nil {
+		converted.OTP = OTPToModel(user.OTP)
+	}
+	return converted
+}
+
+func InitCodeToModel(code *InitUserCode) *model.InitUserCode {
+	return &model.InitUserCode{
+		ObjectRoot: code.ObjectRoot,
+		Expiry:     code.Expiry,
+		Code:       code.Code,
+	}
+}
+
+func (p *User) AppendEvents(events ...*es_models.Event) error {
+	for _, event := range events {
+		if err := p.AppendEvent(event); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (u *User) AppendEvent(event *es_models.Event) error {
+	u.ObjectRoot.AppendEvent(event)
+	var err error
+	switch event.Type {
+	case UserAdded,
+		UserRegistered,
+		UserProfileChanged:
+		u.setData(event)
+	case UserDeactivated:
+		u.appendDeactivatedEvent()
+	case UserReactivated:
+		u.appendReactivatedEvent()
+	case UserLocked:
+		u.appendLockedEvent()
+	case UserUnlocked:
+		u.appendUnlockedEvent()
+	case InitializedUserCodeAdded:
+		u.appendInitUsercodeCreatedEvent(event)
+	case UserPasswordChanged:
+		err = u.appendUserPasswordChangedEvent(event)
+	case UserPasswordCodeAdded:
+		err = u.appendPasswordSetRequestedEvent(event)
+	case UserEmailChanged:
+		err = u.appendUserEmailChangedEvent(event)
+	case UserEmailCodeAdded:
+		err = u.appendUserEmailCodeAddedEvent(event)
+	case UserEmailVerified:
+		u.appendUserEmailVerifiedEvent()
+	case UserPhoneChanged:
+		err = u.appendUserPhoneChangedEvent(event)
+	case UserPhoneCodeAdded:
+		err = u.appendUserPhoneCodeAddedEvent(event)
+	case UserPhoneVerified:
+		u.appendUserPhoneVerifiedEvent()
+	case UserAddressChanged:
+		err = u.appendUserAddressChangedEvent(event)
+	case MfaOtpAdded:
+		err = u.appendOtpAddedEvent(event)
+	case MfaOtpVerified:
+		u.appendOtpVerifiedEvent()
+	case MfaOtpRemoved:
+		u.appendOtpRemovedEvent()
+	}
+	if err != nil {
+		return err
+	}
+	u.ComputeObject()
+	return nil
+}
+
+func (u *User) ComputeObject() {
+	if u.State == 0 {
+		if u.Email != nil && u.IsEmailVerified {
+			u.State = int32(model.USERSTATE_ACTIVE)
+		} else {
+			u.State = int32(model.USERSTATE_INITIAL)
+		}
+	}
+	if u.Password != nil && u.Password.ObjectRoot.AggregateID == "" {
```

Comment:
maybe it makes sense to create a objectRoot.IsZero() method. I it's easier to understand than `u.Password.ObjectRoot.AggregateID == ""`